pyEpiabm/pyEpiabm/sweep/travel_sweep.py

```diff
@@ -214,24 +214,35 @@ def check_leaving_individuals(self, time, person):
         """
         if (hasattr(person, 'travel_end_time')) and \
                 (time > person.travel_end_time):
-            if all([hasattr(person, 'isolation_start_time'),
-                    hasattr(person, 'quarantine_start_time')]):
-                if (person.isolation_start_time is None and
-                        person.quarantine_start_time is None) or \
-                        (person.isolation_start_time <= time and
-                         person.quarantine_start_time <= time):
-                    return True
-            elif hasattr(person, 'isolation_start_time'):
-                if (person.isolation_start_time is None) or \
-                        (person.isolation_start_time <= time):
-                    return True
-            elif hasattr(person, 'quarantine_start_time'):
-                if (person.quarantine_start_time is None) or \
-                        (person.quarantine_start_time <= time):
-                    return True
+            interventions_list = {'isolation_start_time': hasattr(
+                                  person, 'isolation_start_time'),
+                                  'quarantine_start_time': hasattr(
+                                  person, 'quarantine_start_time'),
+                                  'travel_isolation_start_time': hasattr(
+                                  person, 'travel_isolation_start_time')}
+            if any(interventions_list.values()):
+                interventions_not_active = True
+                for intervention in interventions_list.keys():
+                    if interventions_list[intervention]:
+                        if intervention == 'isolation_start_time':
+                            if (person.isolation_start_time is not None) and \
+                                    (person.isolation_start_time <= time):
+                                interventions_not_active = False
+                        if intervention == 'quarantine_start_time':
+                            if (person.quarantine_start_time is not None) and \
+                                    (person.quarantine_start_time <= time):
+                                interventions_not_active = False
+                        if intervention == 'travel_isolation_start_time':
+                            if (person.travel_isolation_start_time is
+                                    not None) and \
+                                    (person.travel_isolation_start_time
+                                     <= time):
+                                interventions_not_active = False
```

Comment:
Could combine these clauses to prevent duplicated code - find the intervention string that contains 'start_time' and then match each intervention string with the associated start time

Reply:
We discussed that we do need `eval()` for this and that that is not good code practice. So we ldo not change this code.